### ISSUE TITLE
[BUG] Backend error while logging in and Signing up

### ISSUE THREAD

**Issue by yashmandi** [Scribbie-Notes/notes-app#119]
steps to reproduce error:
try to signup or login, and open console, you'll get the error description in inspect section.

*important:
after fixing the issue, kindly attach the screenrecording of the solution

![image](https://github.com/user-attachments/assets/780a5c0c-977f-4a18-8b96-67c045a522ef)

![image](https://github.com/user-attachments/assets/227c1a25-e484-43bc-af47-431ed5b4b943)

**Comment by Abhishek332** [comment]
@yashmandi Could I get assign on this issue?, I would love to contribute.

**Comment by zaheer-Khan3260** [comment]
As a Mern Stack developer i have great knowledge in authentcation so please assign too me i love to work on this bug.

**Comment by ishanguptaofficial** [comment]
Hello Scribbie-Notes Team,

My name is Ishan, and I am a 4th-year BTech student with experience in full-stack development. I would love to contribute to your project. Here is my LinkedIn profile: https://www.linkedin.com/in/ishanguptaofficial/.

Could you please assign me to this issue? Looking forward to your response.

Best regards,
Ishan

**Comment by yashmandi** [comment]
you can work on it @Abhishek332

**Comment by Abhishek332** [comment]
Thankyou

**Comment by zaheer-Khan3260** [comment]
Hey, @Abhishek332 If you need any help to solve this problem then Message we can work on it together.

**Comment by RaiDevX8** [comment]
is this issue solved? @yashmandi

**Comment by Abhishek332** [comment]
Hi Guys, wait for the some time, I found the cause, I will be push the fix for it today EOD.
Thanks

**Comment by RaiDevX8** [comment]
nice work @Abhishek332 !!


> Hi Guys, wait for some time, I found the cause, I will be push the fix for it today EOD. Thanks

**Comment by Saurabhchaudhary9799** [comment]
hey @yashmandi , can i work on this issue , because of this issue , i can't work on other issues

**Comment by yashmandi** [comment]
> hey @yashmandi , can i work on this issue , because of this issue , i can't work on other issues

@Abhishek332 is alreay working on this one.

**Comment by yashmandi** [comment]
by when can you get it done @Abhishek332 ?

**Comment by Abhishek332** [comment]
Hi @yashmandi,

I've fixed the issue and am currently doing some testing. While testing, I found a few more bugs and improvements that we can address in the codebase:

We're currently tracking a lot of node_modules packages, which shouldn't be tracked. I've already submitted a PR to fix this. Could you please assign this issue to me under the hacktoberfest label and review/merge the PR? This will help new contributors avoid confusion about why they see many file changes, even though /frontend/node_modules is listed in .gitignore.

Issue Link: #137 
PR Link: [https://github.com/Scribbie-Notes/notes-app/pull/138](https://github.com/Scribbie-Notes/notes-app/pull/138)
I'm getting a `404 error` during login, even though the userData is present in the DB.

After successful signup, all subsequent API requests return a `403 error` (seems like the auth token isn’t being passed for some reason – needs troubleshooting).

I haven’t created separate issues for points 2 and 3 yet. I'll verify these further and then create the necessary issues if needed.

Lastly, I’ve set up a Brevo account and created the `SMTP API key` for sending emails, but I'm unsure what to add for `BREVO_USER` in `backend/.env`. Could you please share the correct value so I can fully test the signup flow for points 2 and 3?

Thanks!

**Comment by Abhishek332** [comment]
> by when can you get it done @Abhishek332 ?

@yashmandi 
https://github.com/user-attachments/assets/57ea2f69-e869-491e-ba7e-953f258d8f54

**Comment by RaiDevX8** [comment]
So now you are on the main page but you cannot create notes it redirects to the login page right @Abhishek332

**Comment by RaiDevX8** [comment]
Am pretty sure login won't work here you done registration only

**Comment by Abhishek332** [comment]
> Am pretty sure login won't work here you done registration only

Yes, I mentioned the login issue in my second point. There seems to be another problem with the login, which results in a 400 (Bad Request) error.
Adding issue link below, could you please assign this issue to me @yashmandi 
[https://github.com/Scribbie-Notes/notes-app/issues/141](https://github.com/Scribbie-Notes/notes-app/issues/141)

Additionally, after registration, all API calls are failing (as noted in my third point). I suspect it might be related to the sendMail process not being completed or verified. I haven't created a ticket for this yet, as it might just be an issue on my end.

**Comment by Abhishek332** [comment]
@RaiDevX8 Could you help me setup `Brevo` for sending email. 
I create a account in Brevo.
Create SMTP secret key
But stuck in what value do I need to add in BREVO_USER in backend/.env

**Comment by tanish35** [comment]
Hi @Abhishek332 
Go to SMTP&API Page there you would see a field called login and a email ends with @smtp-brevo.com. Tho i would say not to use brevo as you also have to set up your custom domain to make it work

https://www.youtube.com/watch?v=kTcmbZqNiGw

Please do watch this video to set up a server directly using gmail. It will be easier also verified and your emails won't go to spam. No need to change the code the details will be the same

**Comment by Abhishek332** [comment]
> Hi @yashmandi,
> 
> I've fixed the issue and am currently doing some testing. While testing, I found a few more bugs and improvements that we can address in the codebase:
> 
> 1. We're currently tracking a lot of node_modules packages, which shouldn't be tracked. I've already submitted a PR to fix this. Could you please assign this issue to me under the hacktoberfest label and review/merge the PR? This will help new contributors avoid confusion about why they see many file changes, even though /frontend/node_modules is listed in .gitignore.
> 
> **Issue Link:** #137 **PR Link:** #138
> 
> 2. I'm getting a `400 (bad request) error` during login
> 3. After successful signup, all subsequent API requests return a `403 error` (seems like the auth token isn’t being passed for some reason – needs troubleshooting).
> 
> I haven’t created separate issues for points 2 and 3 yet. I'll verify these further and then create the necessary issues if needed.
> 
> Lastly, I’ve set up a Brevo account and created the `SMTP API key` for sending emails, but I'm unsure what to add for `BREVO_USER` in `backend/.env`. Could you please share the correct value so I can fully test the signup flow for points 2 and 3?
> 
> Thanks!

@yashmandi @RaiDevX8 Login issue is fixed, please review and merge it
[https://github.com/Scribbie-Notes/notes-app/pull/148](https://github.com/Scribbie-Notes/notes-app/pull/148)

https://github.com/user-attachments/assets/3351b9de-0936-4be3-8aea-a3493feebb66

**Comment by Abhishek332** [comment]
> Hi @yashmandi,
> 
> I've fixed the issue and am currently doing some testing. While testing, I found a few more bugs and improvements that we can address in the codebase:
> 
> 1. We're currently tracking a lot of node_modules packages, which shouldn't be tracked. I've already submitted a PR to fix this. Could you please assign this issue to me under the hacktoberfest label and review/merge the PR? This will help new contributors avoid confusion about why they see many file changes, even though /frontend/node_modules is listed in .gitignore.
> 
> **Issue Link:** #137 **PR Link:** #138
> 
> 2. I'm getting a `400 (bad request) error` during login
> 3. After successful signup, all subsequent API requests return a `403 error` (seems like the auth token isn’t being passed for some reason – needs troubleshooting).
> 
> I haven’t created separate issues for points 2 and 3 yet. I'll verify these further and then create the necessary issues if needed.
> 
> Lastly, I’ve set up a Brevo account and created the `SMTP API key` for sending emails, but I'm unsure what to add for `BREVO_USER` in `backend/.env`. Could you please share the correct value so I can fully test the signup flow for points 2 and 3?
> 
> Thanks!

@yashmandi 3rd one is also fixed. Could you please assign this issue to me under `hacktoberfest` and review and merge it
**issue link** - [https://github.com/Scribbie-Notes/notes-app/issues/149](https://github.com/Scribbie-Notes/notes-app/issues/149)
**Pr Link:** https://github.com/Scribbie-Notes/notes-app/pull/150

**Comment by Abhishek332** [comment]
> Hi @Abhishek332 Go to SMTP&API Page there you would see a field called login and a email ends with @smtp-brevo.com. Tho i would say not to use brevo as you also have to set up your custom domain to make it work
> 
> https://www.youtube.com/watch?v=kTcmbZqNiGw
> 
> Please do watch this video to set up a server directly using gmail. It will be easier also verified and your emails won't go to spam. No need to change the code the details will be the same

Sure, @tanish35 I will definately check it out.
Thankyou

**Comment by RaiDevX8** [comment]
Nice job @Abhishek332

**Comment by Abhishek332** [comment]
@yashmandi This issue can be closed, as all the sub issues I mention in this issue is fixed.
